common/src/main/java/com/regnosys/rosetta/common/transform/TestPackUtils.java

```diff
@@ -81,11 +82,29 @@ public static List<PipelineModel> getPipelineModels(Path resourcePath, ClassLoad
                 .collect(Collectors.toList());
     }
 
-    public static PipelineModel getPipelineModel(List<PipelineModel> pipelineModels, String functionName) {
-        return pipelineModels.stream()
+    //This will return a singular pipeline model with the function name in the list
+    public static List<PipelineModel> getPipelineModel(List<PipelineModel> pipelineModels, String functionName) {
+        List<PipelineModel> filteredModels = pipelineModels.stream()
                 .filter(p -> p.getTransform().getFunction().equals(functionName))
-                .findFirst()
-                .orElseThrow(() -> new IllegalArgumentException(String.format("No PipelineModel found with function name %s", functionName)));
+                .collect(Collectors.toList());
+        if (filteredModels.isEmpty()) {
+            throw new IllegalArgumentException(String.format("No PipelineModel found with function name %s", functionName));
+        }
+        return filteredModels;
+    }
+
+    //This will return a list of pipeline models that match the function name and pipelineId
+    public static List<PipelineModel> getPipelineModel(String pipelineId, List<PipelineModel> pipelineModels, String functionName) {
```

Comment:
Also: can/should there ever be more than one result? If not: it should probably assert there is only one, and just return a single model. (and perhaps be renamed to `findPipelineModel`?) If it should return more than one, it should probably be renamed:
- `getPipelineModels`
- or better: `filterPipelineModels`
- ...

Similar comment for the method above.